lib/remote/endpoint.ti

```diff
@@ -54,6 +54,14 @@ class Endpoint : ConfigObject
 	[no_user_modify, no_storage] double bytes_received_per_second {
 		get;
 	};
+
+	[no_user_modify, no_storage] double seconds_reading_messages {
+		get;
+	};
+
+	[no_user_modify, no_storage] double seconds_processing_messages {
```

Comment:
Colleagues, addition we could also (just for the API) record seconds_processing_messages PER message. I.e. there should be also another attribute returning a dict like {"event::CheckResult":42.0,... saying that this endpoint e.g. spent 42 seconds handling (already read and decoded) event::CheckResult messages since program start.

Reply:
* Yes, there's also #10141, but it's debug log unless a singe message takes excessively long. So, unless combined with #9844, it's pretty useless in a large env.

Reply:
https://github.com/Icinga/icinga2/pull/10141#discussion_r1881689184 legitimately mentions that we already have logs which warn **if the semaphore wait** delays message processing noticeably. I.e you have more busy connections than semaphore slots.

But if you have enough slots and **your messages themselves** take long to process (#10267?), the suggested aggregation tells you.